File: sdk/python/examples/create_sandbox.py
import argparse

from microvm import MicroVM


def main() -> None:
    parser = argparse.ArgumentParser(description="Create a sandbox with the Aerol.ai MicroVM Python SDK")
    parser.add_argument("--api-url", required=False, default="http://127.0.0.1:8080")
    parser.add_argument("--pat-token", required=True)
    parser.add_argument("--image", required=True)
    parser.add_argument("--cpu", type=float, default=1.0)
    parser.add_argument("--memory-mb", type=int, default=1024)
    parser.add_argument("--disk-gb", type=int, default=10)
    args = parser.parse_args()

    client = MicroVM(api_url=args.api_url, pat_token=args.pat_token)
    health = client.health()
    print("health", health)

    sandbox = client.create(
        {
            "image": args.image,
            "cpu": args.cpu,
            "memoryMB": args.memory_mb,
            "diskGB": args.disk_gb,
        }
    )

    print("sandbox", sandbox.to_dict())
    print(f"open {sandbox.publicURL}")


if __name__ == "__main__":
    main()

File: sdk/rust/examples/create_sandbox.rs
use aerolvm_sdk::{Client, CreateOptions};
use std::env;

fn main() {
    let api_url = env::args()
        .nth(1)
        .unwrap_or_else(|| "http://127.0.0.1:21212".to_owned());
    let pat_token = env::args().nth(2).expect("PAT token is required");
    let image = env::args().nth(3).expect("Image is required");

    let client = Client::new(Some(&api_url), Some(&pat_token)).expect("failed to create client");
    let health = client.health().expect("health check failed");
    println!("health = {:#?}", health);

    let sandbox = client
        .create(CreateOptions {
            image,
            cpu: Some(1),
            memory_mb: Some(1024),
            disk_gb: Some(10),
            env: None,
            os_user: None,
            network_block_all: None,
            network_bytes_in_limit: None,
            network_bytes_out_limit: None,
            registry: None,
            container_command: None,
            mounts: None,
            lifecycle: None,
            failover: None,
            runtime: None,
            gpus: None,
            custom_domains: None,
            durability: None,
        })
        .expect("sandbox creation failed");

    println!("sandbox = {:#?}", sandbox.data);
    println!("open {}", sandbox.data.public_url);
}
